### PR TITLE
[feature](inverted index) add slop functionality to match_phrase

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

https://github.com/apache/doris-website/pull/553 doc

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 1]
```diff
@@ -261,14 +263,23 @@ Status FullTextIndexReader::query(OlapReaderStatistics* stats, RuntimeState* run
...
         InvertedIndexQueryCache::CacheKey cache_key;
         auto index_file_key = _inverted_index_file_reader->get_index_file_key(&_index_meta);
 
         if (query_type == InvertedIndexQueryType::MATCH_REGEXP_QUERY) {
             cache_key = {index_file_key, column_name, query_type, search_str};
-            analyse_result.emplace_back(search_str);
+            query_info.terms.emplace_back(search_str);
         } else {
+            if (query_type == InvertedIndexQueryType::MATCH_PHRASE_QUERY) {
+                std::regex slopRegex(R"((.*[^ ])\s~(\d+)$)");
```
[the remark this answers is not in the document]
Comment (zzzxl1993) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.cpp]: changed to not use regex

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**